pandas/core/frame.py

```diff
@@ -3593,14 +3594,27 @@ def transpose(self, *args, copy: bool = False) -> DataFrame:
             and dtypes
             and isinstance(dtypes[0], ExtensionDtype)
         ):
-            # We have EAs with the same dtype. We can preserve that dtype in transpose.
-            dtype = dtypes[0]
-            arr_type = dtype.construct_array_type()
-            values = self.values
+            if isinstance(dtypes[0], BaseMaskedDtype):
+                # We have masked arrays with the same dtype. We can transpose faster.
+                from pandas.core.arrays.masked import transpose_homogenous_masked_arrays
+
+                if isinstance(self._mgr, ArrayManager):
+                    masked_arrays = self._mgr.arrays
+                else:
+                    masked_arrays = [*self._iter_column_arrays()]
```

Comment:
Seems to make no perf difference, but I think `list(self._iter_column_arrays)` is just a bit cleaner. Thoughts?

Reply:
Ok.